### PR TITLE
[LENS-438] Make ModalSurface maxWidth overwrite-able

### PR DESCRIPTION
**Pull request by dbchristopher**
### :sparkles: Changes

-  Adds responsive maxWidth prop support for `ModalSurface` and all consumers

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] PR is ideally < 400LOC

### CODE REVIEW

**Comment by dbchristopher** [comment]
@mdodgelooker Here's my fix for https://looker.atlassian.net/browse/LENS-483 

I added maxWidth prop overrides. Let me know if you think that will be sufficient for the DashboardScene modal that ran into issues last time. No rush of course; happy to sit on this and ship it next week or whatever if you need to focus on summit stuff!

File: packages/www/src/documentation/components/modals/dialog.mdx [hunk 1]
```diff
@@ -49,3 +49,29 @@ DialogManager gives an easy way to compose a Dialog without the need to manage s
   {onClick => <Button onClick={onClick}>Open Dialog </Button>}
 </DialogManager>
 ```
+
+## Props: width and maxWidth
+
+All variants of Dialog allow you to override content `width` and `max-width` style to suit your content. We support responsive width arrays an well as static strings.
```
Comment (lukelooker) [inline packages/www/src/documentation/components/modals/dialog.mdx]: Ok - I had to poke at this to remind myself why we support both of these so I think we should explain them. Here's a stab at it (please feel free to edit my terrible writing):

>
> By default width is unassigned as acts as `auto` sizing the Dialog surface to be wide enough to contain its content. At the same time `maxWidth` constrains the Dialog surface's width to no larger than the specified value.
> 
> `maxWidth` defaults to one of three sizes depending on the responsive breakpoint of the page (`['90vw', '90vw', '600px']`).
> 
> With those constraints in mind if you want a very width Dialog that has complex content you should use `maxWidth` and if have a `Dialog` that must be a specific width assign values to both.

**Comment by dbchristopher** [comment]
ok I updated the docs with Luke's suggestion! I need another ✅whenever you have time

**Comment by dbchristopher** [comment]
@lukelooker had to fix some merge conflicts here, so I'll need another ✅ when you have a moment

**Pull request merged**